src/GitVersion.Core/Configuration/ConfigExtensions.cs

```diff
@@ -7,13 +7,55 @@ namespace GitVersion.Configuration;
 
 public static class ConfigExtensions
 {
-    public static BranchConfig? GetConfigForBranch(this Config config, string? branchName)
+    public static BranchConfig GetBranchConfiguration(this Config configuration, IBranch branch)
+        => GetBranchConfiguration(configuration, branch.NotNull().Name.WithoutRemote);
+
+    public static BranchConfig GetBranchConfiguration(this Config configuration, string branchName)
     {
-        if (branchName == null)
+        var branchConfiguration = FindConfigForBranch(configuration, branchName);
+        if (branchConfiguration is null)
         {
-            throw new ArgumentNullException(nameof(branchName));
+            branchConfiguration = GetUnknownBranchConfiguration(configuration);
+            branchConfiguration.Name = branchName;
         }
+        return branchConfiguration;
+    }
+
+    // TODO: Please make the unknown settings also configurable in the yaml.
+    public static BranchConfig GetUnknownBranchConfiguration(this Config configuration) => new()
+    {
+        Name = "Unknown",
+        Regex = "",
+        Tag = "{BranchName}",
+        VersioningMode = configuration.VersioningMode,
+        Increment = IncrementStrategy.Inherit
+    };
 
+    // TODO: Please make the fallback settings also configurable in the yaml.
+    public static BranchConfig GetFallbackBranchConfiguration(this Config configuration)
```

Comment:
Of course we can do it. What is your recommendation? Do you mean to place it in the yaml file? This will be a breaking change or at least an additional change. I would say we are doing this in v6.